sdk/azcore/internal/pollers/loc/loc.go

```diff
@@ -84,15 +83,22 @@ func (p *Poller[T]) Poll(ctx context.Context) (*http.Response, error) {
 		if h := resp.Header.Get(shared.HeaderLocation); h != "" {
 			p.PollURL = h
 		}
-		p.CurState = resp.StatusCode
+		// if provisioning state is available, use that.  this is only
+		// for some ARM LRO scenarios (e.g. DELETE with a Location header)
+		// so if it's missing then use HTTP status code.
+		provState, _ := pollers.GetProvisioningState(resp)
 		p.resp = resp
-		if p.CurState == http.StatusAccepted {
-			return pollers.StatusInProgress, nil
-		} else if p.CurState > 199 && p.CurState < 300 {
+		if provState != "" {
+			p.CurState = provState
+		} else if resp.StatusCode == http.StatusAccepted {
+			p.CurState = pollers.StatusInProgress
+		} else if resp.StatusCode > 199 && resp.StatusCode < 300 {
 			// any 2xx other than a 202 indicates success
-			return pollers.StatusSucceeded, nil
+			p.CurState = pollers.StatusSucceeded
```

Comment:
Is the comment correct? Do we somehow not land here when `resp.StatusCode == 202`?